code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java

```diff
@@ -349,4 +339,62 @@ private void createTableIfNotExists() {
                         "createTableIfNotExists - Error creating/accessing table (%s)  ",
                         TABLE_NAME));
     }
+
+    /**
+     * Adds a a Table with name {@link #TABLE_NAME}, if not already exists in database at path
+     * {@link #databasePath}.
+     */
+    private boolean tryAddEntity(final DataEntity dataEntity) {
+        return SQLiteDatabaseHelper.process(
+                databasePath,
+                SQLiteDatabaseHelper.DatabaseOpenMode.READ_WRITE,
+                database -> {
+                    if (database == null) {
+                        return false;
+                    }
+                    final int INDEX_UUID = 1;
+                    final int INDEX_TIMESTAMP = 2;
+                    final int INDEX_DATA = 3;
+                    try (SQLiteStatement insertStatement =
+                            database.compileStatement(
+                                    "INSERT INTO "
+                                            + TABLE_NAME
+                                            + " (uniqueIdentifier, timestamp, data) VALUES (?,"
+                                            + " ?, ?)")) {
+                        insertStatement.bindString(INDEX_UUID, dataEntity.getUniqueIdentifier());
+                        insertStatement.bindLong(
+                                INDEX_TIMESTAMP, dataEntity.getTimestamp().getTime());
+                        insertStatement.bindString(
+                                INDEX_DATA,
+                                dataEntity.getData() != null ? dataEntity.getData() : "");
+                        long rowId = insertStatement.executeInsert();
+                        return rowId >= 0;
+                    } catch (Exception e) {
+                        Log.debug(
+                                ServiceConstants.LOG_TAG,
+                                LOG_PREFIX,
+                                "add - Returning false: " + e.getLocalizedMessage());
+                        return false;
+                    }
+                });
+    }
+
+    private void resetDatabase() {
+        Log.warning(
+                ServiceConstants.LOG_TAG,
+                LOG_PREFIX,
+                "resetDatabase - Resetting database (%s) as it is corrupted",
+                databasePath);
+
+        try {
+            FileUtils.deleteFile(new File(databasePath), false);
+            createTableIfNotExists();
```

Comment:
Do you need to synchronize on the `dbMutex` here?

Reply:
Ah, I see that you are. Can you add a comment to this method stating the caller must hold the `dbMutex`?